src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs

```diff
@@ -1,136 +1,1540 @@
+
+        // Converters
+        private static InverseBooleanToVisibilityCollapsedConverter _inverseBooleanToVisibilityCollapsedConverter = new InverseBooleanToVisibilityCollapsedConverter();
+        private static BoolToVisibilityCollapsedConverter _boolToVisibilityCollapsedConverter = new BoolToVisibilityCollapsedConverter();
+        private static BoolToVisibilityConverter _booleanToVisibilityConverter = new BoolToVisibilityConverter();
+        private static EmptyToVisibilityCollapsedConverter _emptyToVisibilityCollapsedConverter = new EmptyToVisibilityCollapsedConverter();
+        private static ZoomToVisibilityCollapsedConverter _zoomToVisibilityCollapsedConverter = new ZoomToVisibilityCollapsedConverter();
+
+        // Font
+        private static FontFamily _artifactElementReg = SharedDictionaryManager.DynamoModernDictionary["ArtifaktElementRegular"] as FontFamily;
+
+        // Images
+        private static readonly BitmapImage _frozenImageSource = new BitmapImage(new Uri("pack://application:,,,/DynamoCoreWpf;component/UI/Images/NodeStates/frozen-64px.png"));
+        private static readonly BitmapImage _transientImageSource = new BitmapImage(new Uri("pack://application:,,,/DynamoCoreWpf;component/UI/Images/NodeStates/transient-64px.png"));
+        private static readonly BitmapImage _hiddenEyeImageSource = new BitmapImage(new Uri("pack://application:,,,/DynamoCoreWpf;component/UI/Images/hidden.png"));
+        private static readonly BitmapImage _nodeButtonDotsSelected = new BitmapImage(new Uri("pack://application:,,,/DynamoCoreWpf;component/UI/Images/more-vertical_selected_16px.png"));
+        private static readonly BitmapImage _nodeButtonDots = new BitmapImage(new Uri("pack://application:,,,/DynamoCoreWpf;component/UI/Images/more-vertical.png"));
+        private static ImageBrush _defaultNodeIcon = new ImageBrush(new BitmapImage(new Uri("pack://application:,,,/DynamoCoreWpf;component/UI/Images/default-node-icon.png")))
+        {
+            Stretch = Stretch.UniformToFill
+        };
+
+        private static Style _zoomFadeOpacity_OneToZeroStyle = GetZoomFadeOpacity_OneToZeroAnimatedStyle();
+        private static Style _zoomFadeOpacity_50PercentToZeroStyle = GetZoomFadeOpacity_50PercentToZeroAnimatedStyle();
+        private static Style _zoomFadeInOpacity_ZeroTo50PercentStyle = GetZoomFadeInOpacity_ZeroTo50PercentAnimatedStyle();
+        private static Style _nodeButtonStyle = GetNodeButtonStyle();
+        private static Style _codeBlockNodeItemControlStyle = GetCodeBlockPortItemControlStyle();
+        internal static readonly Style DynamoToolTipTopStyle = GetDynamoToolTipTopStyle();
+        private static ContextMenu nodeContextMenu = GetNodeContextMenu();
+
+        #region constructors
+        static NodeView()
+        {
+            //Freeze the static resource to reduce memory overhead
+            _frozenImageSource.Freeze();
```

Comment:
Is this called during initilaization, I am asking because there is another default public constructor for NodeView and freezing it would make it immutable so we should be setting all the properties and then freezing the images.

Reply:
We are only assigning the _frozenImageSource as the source for the FrozenImageSource, not actually changing any properties for _frozenImageSource